### DISCUSSION TITLE
[Feature] Add a copy button for the share link on localhost urls

### DISCUSSION THREAD

**Discussion by tranquanghuy0801** [immich-app/immich#3393] [Feature Request]
### The feature

After creating the link to share, the copy button at the end will be great to help the user to copy the link

![Screen Shot 2023-07-23 at 9 58 25 pm](https://github.com/immich-app/immich/assets/29462498/aaf1f86f-9009-48c3-b07b-cfbf98fd5286)

### Platform

- [ ] Server
- [X] Web
- [ ] Mobile

**Comment by alextran1502** [comment]
The copy button will appear if you are accessing the site via https. Http doesn't allow you to access clipboard due to security reasons

**Reply by tranquanghuy0801** [reply]
Oh I did not know it. Thanks for the info

**Reply by jrasm91** [reply]
We should actually just check if the clipboard is available. Localhost is a secure context, even without https.

> Locally-delivered resources such as those with http://127.0.0.1 URLs, http://localhost and http://*.localhost URLs (e.g. http://dev.whatever.localhost/), and file:// URLs are also considered to have been delivered securely.

**Reply by martabal** [reply]
If you use firefox you have to set `dom.events.asyncClipboard.clipboardItem` to `true` in your `about:config`. 
https://www.npmjs.com/package/copy-image-clipboard

It's disabled by default, I didn't know Immich had this feature before your comment.

**Reply by tranquanghuy0801** [reply]
Actually, I tested on Chrome and the Copy button works fine for localhost. Only Firefox has this issue. Do we still need a fix for this?
